### PR TITLE
Rendered the list to the list page

### PR DESCRIPTION
**Pull request by KarenJParker77**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### CODE REVIEW

**Comment by mikeyxx** [comment]
Hi Karen and Cassandra... Sandra and I have checked your code and it looks great to us. We wouldn't change anything!

File: src/views/List.jsx [hunk 1]
```diff
@@ -6,12 +6,11 @@ export function List({ data }) {
...
 				Hello from the <code>/list</code> page!
 			</p>
+
 			<ul>
-				{/**
-				 * TODO: write some JavaScript that renders the `data` array
-				 * using the `ListItem` component that's imported at the top
-				 * of this file.
-				 */}
+				{data.map((data) => (
```
Comment (edieblu) [inline src/views/List.jsx]: To make this easier to read and avoid confusion, it's best to distinguish between the array (`data`) and its values (also called `data` here). 😊

Often the convention is going from plural to singular, for example, if the array is called `books`, then the item would be `book`. Here you could use something like `item`.

File: src/views/List.jsx [hunk 2]
```diff
@@ -6,12 +6,11 @@ export function List({ data }) {
...
+					<ListItem name={data.name} />
```
Comment (edieblu) [inline src/views/List.jsx]: When mapping over an array, you need to give each array item a key 😊
There's an error in the console that warns you about that.
image

You can read more about it [here](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key).

**Review by edieblu — CHANGES_REQUESTED** [review]
Hello team!

Good job on submitting your first PR! 👏🥳

Before we merge this, we'll need to fix a couple of things. Please go over the comments and let me know if you have any questions. 😊 Once you're ready please request a re-review. Thanks!

**Comment by mikeyxx** [comment]
Hi Karen and Cassandra, your code works fine, to make it more efficient, I think it will be better to use an 'id' as your key. Using the name of the product is fine as there may not be more than one of a product type on the list but I think it is safer to use the product id for the key to help React better identify each item as unique.

If you log out the 'data' value in the console, you will see the data returned has an id value. 


Screenshot 2023-04-06 091609


You can use the 'id' value as the key in the list component

File: package.json [hunk 3]
```diff
@@ -9,7 +9,8 @@
 		"firebase": "^9.17.2",
 		"react": "^18.2.0",
 		"react-dom": "^18.2.0",
-		"react-router-dom": "^6.9.0"
+		"react-router-dom": "^6.9.0",
+		"uuid": "^9.0.0"
```
Comment (edieblu) [inline package.json]: I think we can remove this 😊

File: src/views/List.jsx [hunk 4]
```diff
@@ -6,12 +6,11 @@ export function List({ data }) {
...
 			</p>
+
 			<ul>
-				{/**
-				 * TODO: write some JavaScript that renders the `data` array
-				 * using the `ListItem` component that's imported at the top
-				 * of this file.
-				 */}
+				{data.map((item) => (
+					<ListItem key={item.id} name={item.name} />
```
Comment (edieblu) [inline src/views/List.jsx:12]: Yes, exactly 👍 Nicely done!

**Review by edieblu — APPROVED** [review]
Great job! 

Please remove the uuid library, and then this is ready to be merged! 🚀

**Comment by edieblu** [comment]
> Hi Karen and Cassandra, your code works fine, to make it more efficient, I think it will be better to use an 'id' as your key. Using the name of the product is fine as there may not be more than one of a product type on the list but I think it is safer to use the product id for the key to help React better identify each item as unique.
> 
> If you log out the 'data' value in the console, you will see the data returned has an id value.
> 
> Screenshot 2023-04-06 091609
> 
> You can use the 'id' value as the key in the list component

Thank you @mikeyxx ! 👏

**Pull request merged**